### DISCUSSION TITLE
發表文章的獎勵方式

### DISCUSSION THREAD

**Discussion by gasolin** [lun-dao/LunDAO#40] [General]
@yurenju 在 #35 提到：
> 固定一段時間後比如說一個月或一季，看看用什麼適合的方式來讓大家討論哪篇文章對社群影響力最大。目前的想法是會有一個固定的文章獎勵跟加成的獎勵，比如說 100 LUN (固定) + 0-100 LUN (浮動)

@chnejohnson 在 https://github.com/lun-dao/LunDAO/discussions/35#discussioncomment-1922993 提到
可以用平方投票法的方式對文章進行捐款（投票）和獎勵分配 
並於 https://github.com/lun-dao/LunDAO/discussions/35#discussioncomment-1923329 進一步作解釋

>

**Comment by gasolin** [comment]
因為我們已預計採用社群貢獻分配 (coordinape 互投) 的方式，加上另外有的 snapshot 投票，若再加入 clr.fund 是不是就有三種投票機制了 XD
@yurenju @hydai  coordinape 或 snapshot 也能用平方投票法嗎？

若是簡化成 

> 保底 (ex: 固定稿費 100 LUN) + 定期社群貢獻分配 (coordinape 互投) 

理想情況下，若文章熱門或極有貢獻，社群貢獻分配這塊應該也可以得到不錯的% ?
 
想到的缺點是若發布的區間有多篇發布，這些作者也只能透過社群貢獻分配

**Reply by yurenju** [reply]
我覺得是只有前兩種方式：
- coordinape: 就直接用 coordinape 就好
- clr.fund: 應該是用 clr 投票，然後用 Disperse.app 發送獎勵
- snapshot: 我覺得應該用在別的地方，不會用在文章分配獎勵

我覺得我們先決定前兩項要用哪項好了， snapshot 先觀察看看要在哪邊用比較適合。比如說我們先測試 coordinape 用在一般貢獻的分配，然後 clr 用在文章貢獻分配。兩邊都試過之後，大家再來討論一下這兩種分配的優缺點是什麼。

**Reply by yurenju** [reply]
> 想到的缺點是若發布的區間有多篇發布，這些作者也只能透過社群貢獻分配

這個雖然有一個保底的貢獻，不過彈性的獎勵分配部分確實有你講的問題，我暫時沒有想出其他好的解決方案。

**Comment by johnson86tw** [comment]
我也覺得 coordinape 適合用在一般貢獻的分配，clr.fund 則用在文章貢獻的分配。

文章貢獻的部分我覺得可以先採用「固定稿費」為主，例如投稿一篇就給 100 LUN；clr.fund 可以作為定期舉辦的額外獎勵活動，例如每月舉辦一次，蒐集該月有哪些作者貢獻了哪些文章，然後讓社群用 LUN 投票（捐款），DAO 提供額外的獎助金，透過平方募資把獎助金分配給該月受歡迎的作者們。

**Reply by yurenju** [reply]
@chnejohnson  想要請你登錄你的 ethereum account 到 [contributors.csv][1]，謝囉！

[1]: https://github.com/lun-dao/LunDAO/blob/main/contributors.csv

**Comment by hydai** [comment]
我在思索是不是可以從 ethereum 現行的 gas fee 模型加上社群投票為主：
1. base(投稿就有基本稿費)
2. priority(有人想看可以直接加碼懸賞) 
3. community vote(社群投票，可能用平方投票法，給額外獎勵) 
4. flow(根據文章的熱門度，給額外獎勵)

3 -> 社群認為高品質的作品
4 -> 受眾認為高品質的作品

**Reply by yurenju** [reply]
覺得這樣滿好的，1 跟 3 我們應該有既有的方式可以做。2 可以先用基本的流程來 cover 之後再看需要來寫 smart contract。目前想像起來應該是：

1. 想看文章的人在 github 上面發一個 issue，標籤選 "article wanted"，比如說 issue id 是 123
2. 兩種方式
    1. 有一個 smart contract 有一個 function articleWanted(issueId, tokenAmount)，呼叫他之後會把 LUN 列到懸賞獎金裡面
    2. 簡單暴力的方法是直接把 LUN 轉到 gnosis safe，然後自己提供 tx id 到 issue 上 <--- 剛開始可以用這個
3. 有人看到懸賞後認領
4. 文章寫好發布後，把 LUN token 轉給他

至於 community vote 就照 @hydai 跟 @chnejohnson 討論的方式用平方投票法。4 我們可以等到有流量統計的時候再做，一步步來。我想要找一個 privacy first 的流量統計方式。
